libpimeval/src/pimSim.cpp

```diff
@@ -41,13 +46,54 @@ pimSim::destroy()
 //! @brief  pimSim ctor
 pimSim::pimSim()
 {
-  m_paramsDram = new pimParamsDram();
-  m_paramsPerf = new pimParamsPerf(m_paramsDram);
-  m_statsMgr = new pimStatsMgr(m_paramsDram, m_paramsPerf);
 }
 
 //! @brief  pimSim dtor
 pimSim::~pimSim()
+{
+  uninit(); 
+}
+
+//! @brief  Initialize pimSim member classes from the config file
+void
+pimSim::init(const std::string& simConfig)
+{
+  if (!m_initCalled) {
+    bool success = parseConfigFromFile(simConfig);
+    assert(success);
+
+    if (m_memConfigFileName != m_defaultMemConfig) {
+      std::string memConfigFileFullPath = m_configFilesPath + m_memConfigFileName;
+      std::string fileContent;
+      success = pimUtils::readFileContent(memConfigFileFullPath.c_str(), fileContent);
+      assert(success);
+      m_paramsDram = new pimParamsDram(fileContent);
+    }
+    else {
```

Comment:
Minor coding style: Suggest putting else inline for consistency and compactness, e.g. "} else {"

Reply:
Thanks, Fixed these issues.